src/auth/auth.service.ts

```diff
@@ -0,0 +1,148 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+import { Response } from 'express';
+import { DataSource } from 'typeorm';
+import { AuthLoginDto } from './dto/auth-login.dto';
+import { Admin } from '../admin/admin.entity';
+import { Student } from '../student/student.entity';
+import { Recruiter } from '../recruiter/recruiter.entity';
+import { hashPwd } from '../utils/hash-pwd';
+import { v4 as uuid } from 'uuid';
+import { JwtPayload } from './strategy/admin.strategy';
+import { sign } from 'jsonwebtoken';
+import { configCookie } from '../config/configCookie.example';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @Inject(forwardRef(() => DataSource)) private dataSource: DataSource,
+  ) {}
+
+  async login(req: AuthLoginDto, res: Response): Promise<any> {
+    let user: Admin | Student | Recruiter | null = null;
+    try {
+      const admin = await Admin.findOne({
+        where: {
+          email: req.email,
+          pwdHash: hashPwd(req.pwd),
+        },
+      });
+      if (!admin) {
+        const student = await this.dataSource
+          .getRepository(Student)
+          .createQueryBuilder('student')
+          .leftJoinAndSelect('student.studentImport', 'studentImport')
+          .where('studentImport.email = :email', {
+            email: req.email,
+          })
+          .andWhere('student.pwdHash = :pwdHash', {
+            pwdHash: hashPwd(req.pwd),
+          })
+          .andWhere('studentImport.isActive = :active', {
+            active: true,
+          })
+          .getOne();
+        if (!student) {
+          const recruiter = await Recruiter.findOne({
+            where: {
+              email: req.email,
+              pwdHash: hashPwd(req.pwd),
+              isActive: true,
+            },
+          });
+          if (!recruiter) {
+            return res.json({
+              error: 'Invalid login data or account not active!',
+            });
+          } else {
```

Comment:
Ło Panie :D To trzeba uprościć - np. odwrócić tę logikę

Reply:
To fakt :) Nie wygląda to ładnie. 
Najpierw skupiłem się na tym aby działało, a potem refactoring